### DISCUSSION TITLE
Cant Display Thumbnail

### DISCUSSION THREAD

**Discussion by john-helyar** [mriscoc/Ender3V2S1#113] [General]
Hi There

I have updated to the latest Firmware for my Printer and it doesn't display the Thumbnail I get the stats and info but no image I have installed the python file and added the script and the sizes are fine works fine on my other printer no problem and display's fine.

I have also tried to flash the Screen with a new DWIN_SET Folder to have different icons and that doesn't seem to take either

Can you help, please

**Comment by mriscoc** [comment]
Maybe you have a DACAI screen.  https://github.com/mriscoc/Ender3V2S1/wiki/How-to-update-the-display

**Reply by john-helyar** [reply]
Ok so i need to copy the Private folder instead to my SD Card and try using that?

**Comment by john-helyar** [comment]
Ok so i was able to flash the screen using the Private folder and it has now all updated to your screen but Thumbnails still don't seem to want to display

**Reply by mriscoc** [reply]
Please read the Wiki https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview

**Comment by john-helyar** [comment]
Yea i have done all of that hence it works fine on my other printer I have the Python File installed and it is set to the default sizes which match the images

**Reply by mriscoc** [reply]
![image](https://user-images.githubusercontent.com/2745567/162560762-75f237d5-eeda-4086-934e-beac77690a69.png)

**Comment by john-helyar** [comment]
Bugger well that is a pain in the butt then oh well would of been nice maybe at some point I will replace it and go from there 

Cheers for your help on this though.

**Comment by swordstudiox** [comment]
Is it possible to fix DACAI screen problem?
Where is the DACAI screen firmware source code?

**Reply by mriscoc** [reply]
Creality must to fix that, DWIN and DACAI firmware are closed source.

**Reply by Victordiazbis** [reply]
So we can't do anything but wait. Is that right?

**Comment by jxer** [comment]
is this suppose to generate a .jpg file on the sd card? I followed all the steps, but there is no jpg generated in cura 5.0,  and followed all steps

**Reply by mriscoc** [reply]
The JPEG image is embedded in to the G-code file.

**Comment by BscottJ** [comment]
I have been trying to get Cura to add the proper code to the file. The sample file shows the preview just fine but all of the files I create give the QR code instead of a preview of the file

**Reply by Triffen** [reply]
Are you using 'Create JPEG thumbnail' in Cura?
![image](https://user-images.githubusercontent.com/4696816/199562013-da6ff581-f7c5-4157-86eb-b319381fd84c.png)

**Reply by BscottJ** [reply]
I cannot get the Create Jpeg Thumbnail to show up in the choices, using Cura 5.2.1 and it only shows create Thumbnail, how can I make this work or do I go back to an older version of Cura?

**Reply by Triffen** [reply]
You have to install it first like Wiki says. 
![image](https://user-images.githubusercontent.com/4696816/199996336-c8d6e04c-9110-4c6e-9128-c2aaf6a663be.png)

**Reply by BscottJ** [reply]
I HAVE done these steps but the Create Jpeg Thumbnail does not show up

**Reply by Triffen** [reply]
Ah, OK. I'd think either the location is wrong for your slicer, or it doesn't like version of Python in the system. The latter I'm not qualified to troubleshoot, sorry.

**Comment by mriscoc** [comment]
Follow the troubleshooting section in the Wiki: https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview#troubleshootinghttps://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview#troubleshooting

Paste here the results of each test.

**Reply by BscottJ** [reply]
the scripts folder had an extra folder inside __pycache__ with a compiled script that was not letting the file show up. I removed that file and now the create Jpeg script shows up. I will update how it worked after the current print is finished
